sdk/samples/iot/sample_sas_utility.c

```diff
@@ -28,15 +28,15 @@ az_result sample_base64_decode(az_span base64_encoded, az_span in_span, az_span*
   b64_decoder = BIO_new(BIO_f_base64());
   if (b64_decoder == NULL)
   {
-    return AZ_ERROR_OUT_OF_MEMORY;
+    return AZ_ERROR_HTTP_ADAPTER_OUT_OF_MEMORY;
   }
 
   // Get the source BIO to push through the filter
   source_mem_bio = BIO_new_mem_buf(az_span_ptr(base64_encoded), (int)az_span_size(base64_encoded));
   if(source_mem_bio == NULL)
   {
     BIO_free(b64_decoder);
-    return AZ_ERROR_OUT_OF_MEMORY;
+    return AZ_ERROR_HTTP_ADAPTER_OUT_OF_MEMORY;
```

Comment:
This doesn't make sense. It is OK to keep `AZ_ERROR_OUT_OF_MEMORY` for this purpose. Do you have other ideas for this? Maybe this belongs as a core facility rather than platform.

Reply:
There are two trains of thought here:
1) Since Core itself never allocates, it will never return this az_result and using that facility gives the false impression that some allocations are going on. So, don't keep it in core, maybe platform?
2) Since this is a very common/general purpose error, it is possible that upstreamconsumers would use it for things totally unrelated to the hardware/device platform (for example, see [this](https://expresslogic.visualstudio.com/X-Ware/_git/netx?path=%2Faddons%2Fazure_iot%2Fnx_azure_iot_pnp_client.c&version=GBdevelop_higupt_pnp_client&line=16&lineEnd=17&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)). So, don't keep it in platform.

So, there is no clear answer. Therefore, I am OK with either. We could potentially add a separate facility for upstream consumers, with a general purpose SDK facility.

@JeffreyRichter - what do you think?

Reply:
Out of memory should **not** be specific to the adapter.  It is a CORE error code

Reply:
Agree with @ahsonkhan ,  CORE never allocates.  It makes no sense to have an error of `out of memory`. It brings confusion.

the platform facility should be keep, IMO

Reply:
Sorry to jump in here.... but IMO:

while core itself doesn't allocate memory, it seems to me the intention of "core" error codes are ones that may be used by multiple services so we don't have duplicates. There seems to be a decent chance of other things allocating memory in the future in which case, there could be an out of memory issue. We wouldn't want multiple error codes which do the same thing would we?